lib/simplecov/source_file.rb

```diff
@@ -149,46 +246,27 @@ def relevant_lines
       lines.size - never_lines.size - skipped_lines.size
     end
 
-    # Returns all covered lines as SimpleCov::SourceFile::Line
-    def covered_lines
-      @covered_lines ||= lines.select(&:covered?)
+    def no_branches?
+      total_branches.length.zero?
     end
 
-    # Returns all lines that should have been, but were not covered
-    # as instances of SimpleCov::SourceFile::Line
-    def missed_lines
-      @missed_lines ||= lines.select(&:missed?)
-    end
+    def branches_coverage_percent
+      return 100.0 if no_branches? && no_lines?
```

Comment:
still not sure why we need to check `no_lines?` :)

Reply:
Yep, It should be removed, I will do it!